docs/integrations/sources/hubspot.md

```diff
@@ -174,48 +198,79 @@ Then, go to the replication settings of your connection and click **refresh sour
 
 Because of this, the `engagements` stream can be slow to sync if it hasn't synced within the last 30 days and/or is generating large volumes of new data. We therefore recommend scheduling frequent syncs.
 
-## Performance considerations
+## Limitations & Troubleshooting
 
-The connector is restricted by normal HubSpot [rate limitations](https://legacydocs.hubspot.com/apps/api_guidelines).
+<details>
+<summary>
+Expand to see details about Hubspot connector limitations and troubleshooting.
+</summary>
 
-Some streams, such as `workflows`, need to be enabled before they can be read using a connector authenticated using an `API Key`. If reading a stream that is not enabled, a log message returned to the output and the sync operation only sync the other streams available.
+### Connector limitations
 
-Example of the output message when trying to read `workflows` stream with missing permissions for the `API Key`:
+#### Rate limiting
 
-```text
-{
-    "type": "LOG",
-    "log": {
-        "level": "WARN",
-        "message": 'Stream `workflows` cannot be proceed. This API Key (EXAMPLE_API_KEY) does not have proper permissions! (requires any of [automation-access])'
-    }
-}
-```
+The connector is restricted by normal HubSpot [rate limitations](https://legacydocs.hubspot.com/apps/api_guidelines).
 
-HubSpot's API will [rate limit](https://developers.hubspot.com/docs/api/usage-details) the amount of records you can sync daily, so make sure that you are on the appropriate plan if you are planning on syncing more than 250,000 records per day.
+### Troubleshooting
 
-## Tutorials
+* Consider checking out the following Hubspot tutorial: [Build a single customer view with open-source tools](https://airbyte.com/tutorials/single-customer-view).
+* **Enabling streams:** Some streams, such as `workflows`, need to be enabled before they can be read using a connector authenticated using an `API Key`. If reading a stream that is not enabled, a log message returned to the output and the sync operation only sync the other streams available.
```

Comment:
Some streams, such as `workflows`, need to be enabled before they can be read by a connector authenticated using an `API Key`. If reading of a stream is not enabled, a log message will be returned to the output and the sync operation will only sync the other streams available.